h/views/client.py

```diff
@@ -24,36 +23,34 @@
 DEFAULT_CLIENT_URL = 'https://cdn.hypothes.is/hypothesis'
 
 
-def url_with_path(url):
-    if urlparse.urlparse(url).path == '':
-        return '{}/'.format(url)
-    else:
-        return url
-
-
 def _client_url(request):
     """
     Return the configured URL for the client.
     """
     return request.registry.settings.get('h.client_url', DEFAULT_CLIENT_URL)
 
 
-def _app_html_context(api_url, service_url, sentry_public_dsn,
-                      websocket_url, auth_domain, ga_client_tracking_id,
-                      client_url):
+@view_config(route_name='sidebar_app',
+             renderer='h:templates/app.html.jinja2')
+def sidebar_app(request, extra=None):
     """
-    Returns a dict of asset URLs and contents used by the sidebar app
-    HTML tempate.
+    Return the HTML for the Hypothesis client's sidebar application.
```

Comment:
Don't worry about it here but I think being more careful about these names may be something to discuss in this client technical debt meeting. In the code (mostly in the client code but also sometimes in h) we have multiple different things called app, and multiple called sidebar, and now here is "sidebar app" :)

I think there are at least three distinct things: **1.** The `embed.js` script when it's used in the parent document (called `src/annotator/` in the client repo) **2.** The `app.html` page that contains the `<hypothesis-app>`, a `"js-hypothesis-config"` object, and the `<script src="/embed.js">` **3.** The actual AngularJS app that's delivered in `embed.js` (the second time that script is run) and rooted at the `<hypothesis-app>` element. There may be more bits that I'm forgetting. But we need to come up with distinct names for each component and then use them consistently.